app/config_tools.py

```diff
@@ -1,163 +1,242 @@
-        
-        print("Attempting to find posters directory")
-        
-        if 'poster-directory' in config:
-            self.posterdirectory = config['poster-directory']
-        else:
-            app_dir = os.path.dirname(os.path.realpath(__file__))
-
-            # Test separate config directory with nested 'posters' directory
-            if os.path.exists(os.path.join(app_dir, "..", "config", "posters")):
-                self.posterdirectory = os.path.abspath(os.path.join(app_dir, "..", "config", "posters"))
-            # Test separate config folder with nested 'images' directory
-            elif os.path.exists(os.path.join(app_dir, "..", "config", "images")):
-                self.posterdirectory = os.path.abspath(os.path.join(app_dir, "..", "config", "images"))
-            # Test nested posters directory
-            elif os.path.exists(os.path.join(app_dir, "posters")):
-                self.posterdirectory = os.path.abspath(os.path.join(app_dir, "posters"))
-            # Test nested images directory
-            elif os.path.exists(os.path.join(app_dir, "images")):
-                self.posterdirectory = os.path.abspath(os.path.join(app_dir, "images"))
-            else:
-                print("Invalid poster-directory setting")
-            
-        if self.posterdirectory: 
-            print("Using {} as poster directory".format(self.posterdirectory))
-        
-        print("Attempting to find backgrounds directory")
-        
-        if 'background-directory' in config:
-            self.backgrounddirectory = config['background-directory']
+
+        if config:
+            self.poster = config['poster-directory'] if 'poster-directory' in config and config['poster-directory'] and os.path.exists(os.path.abspath(config['poster-directory'])) else None
+            self.background = config['background-directory'] if 'background-directory' in config and config['background-directory'] and os.path.exists(os.path.abspath(config['background-directory'])) else None
+            self.image = config['image-directory'] if 'image-directory' in config and config['image-directory'] and os.path.exists(os.path.abspath(config['image-directory'])) else None
         else:
-            app_dir = os.path.dirname(os.path.realpath(__file__))
-
-            # Test separate config directory with nested 'posters' directory
-            if os.path.exists(os.path.join(app_dir, "..", "config", "backgrounds")):
-                self.backgrounddirectory = os.path.abspath(os.path.join(app_dir, "..", "config", "backgrounds"))
-            # Test nested posters directory
-            elif os.path.exists(os.path.join(app_dir, "backgrounds")):
-                self.backgrounddirectory = os.path.abspath(os.path.join(app_dir, "backgrounds"))
+            self.poster = "posters" if os.path.exists(os.path.abspath("posters")) else "..\\config\\posters" if os.path.exists(os.path.abspath("..\\config\\posters")) else None
```

Comment:
I'm think `os.path` resolves based on the working directory instead of the script directory. I think we'll need to include `os.path.dirname(os.path.realpath(__file__))` as the base path on all of these tests. Otherwise, there will be different results if users run the script from `./plex-auto-collections` vs. `./plex-auto-collections/app`.

Reply:
I'm very new to python so I'm doing a lot of guesswork lol. I was only ever running from inside the app folder but ill look into it